Rubberduck.Inspections/Concrete/UnreachableCaseInspection.cs

```diff
@@ -0,0 +1,1309 @@
+            {
+                return SetTheTypeNames(selectStmtDO, typeName);
+            }
+            selectStmtDO.CanBeInspected = false;
+            return selectStmtDO;
+        }
+
+        private SelectStmtDataObject InitializeCaseClauses(SelectStmtDataObject selectStmtDO)
+        {
+            for (var idx = 0; idx < selectStmtDO.CaseClauseDOs.Count; idx++)
+            {
+                var caseClauseDO = selectStmtDO.CaseClauseDOs[idx];
+                if (caseClauseDO.ResultType != ClauseEvaluationResult.Unreachable)
+                {
+                    for (var rgIdx = 0; rgIdx < caseClauseDO.RangeClauseDOs.Count; rgIdx++)
+                    {
+                        var rgClause = caseClauseDO.RangeClauseDOs[rgIdx];
+                        rgClause.CanBeInspected = ContextCanBeEvaluated(rgClause.Context, selectStmtDO.IdReferenceName);
+                        if (rgClause.CanBeInspected)
+                        {
+                            rgClause = InitializeRangeClauseDataObject(rgClause, selectStmtDO.BaseTypeName, selectStmtDO.IdReferenceName);
+                        }
+                        caseClauseDO.RangeClauseDOs[rgIdx] = rgClause;
+                    }
+                }
+                selectStmtDO.CaseClauseDOs[idx] = caseClauseDO;
+            }
+            return selectStmtDO;
+        }
+
+        private SelectStmtDataObject InspectSelectStmtCaseClauses(SelectStmtDataObject selectStmtDO)
+        {
+            for (var idx = 0; idx < selectStmtDO.CaseClauseDOs.Count(); idx++)
+            {
+                var caseClause = selectStmtDO.CaseClauseDOs[idx];
+
+                if (selectStmtDO.SummaryCaseClauses.CaseElseIsUnreachable)
+                {
+                    caseClause.ResultType = ClauseEvaluationResult.Unreachable;
+                    selectStmtDO.CaseClauseDOs[idx] = caseClause;
+                    continue;
+                }
+
+                caseClause.ResultType = caseClause.RangeClauseDOs.All(rg => selectStmtDO.SummaryCaseClauses.RangeClausesAsText.Contains(rg.AsText)) ?
+                    ClauseEvaluationResult.Unreachable : caseClause.ResultType;
+
+                if (caseClause.ResultType == ClauseEvaluationResult.NoResult)
+                {
+                    caseClause = InspectCaseClause(caseClause, selectStmtDO.SummaryCaseClauses);
+                    selectStmtDO = UpdateSummaryClauses(selectStmtDO, caseClause);
+
+                    if (caseClause.RangeClauseDOs.All(rg => rg.ResultType != ClauseEvaluationResult.NoResult))
+                    {
+                        caseClause.ResultType = caseClause.RangeClauseDOs.All(rg => rg.ResultType == ClauseEvaluationResult.Unreachable)
+                            ? ClauseEvaluationResult.Unreachable : caseClause.RangeClauseDOs.All(rg => rg.ResultType == ClauseEvaluationResult.MismatchType)
+                                ? ClauseEvaluationResult.MismatchType : ClauseEvaluationResult.NoResult;
```

Comment:
Could this be more explicit and less ternaries, please? I find it hard to follow this

Reply:
Modified...yes...more readable.